### PR TITLE
Light Emote Refactor and Heavy Emote Expansion

### PR DESCRIPTION
**Pull request by GrrrKitten**
# About the pull request

First thing: let it be known I tested all of these extensively, this PR does not make the game spammy, there are just many pick options and a lot of new emotes added.

---

Edit: putting this here so I don't forget for later TODO- THE XENO THING YOU FORGOT 

Major Changes:
- If you are deaf but the emote is a visible emote, you can now see it
- If you are blind but the emote is audible, you can now see it
- If you are blind and someone talks, you can now see it (aka blind people are no longer also deaf)
- Adds a brand new section of non-player useable emotes
- Adds a wide range of emotes to things that didn't emote before for player QoL and RP experience

Major Specifics:
- CPR'ing now emotes who youre CPR'ing
- Defibbing now emotes who youre defibbing
- Surgery now emotes what you did when you complete a step
- Splinting now emotes what limb youre splinting on who
- Organ damages now give visible emotes allowing you to attempt to go Dr House mode and prescribe your fellow RFN more mouse bites
- Dying now emotes your death message
- Boiler neuro gas is now packed with emotes making it absolutely brutal, turns CM into a true warzone 

Minor Changes:
- Changes a lot of emotes that were incorrectly marked as not-doable while unconcious
- Changes a lot of emotes that were incorrectly marked with as audible/visible emotes

Minor Specifics:
- Stripping/placing items on players now emotes
- Cuffing players now emotes
- Going in vents now emotes

# Explain why it's good for the game

Emotes exist to allow players to understand whats going on around them when a do_after circle isnt enough and chat is moving fast. Emotes add a lot to the RP and the intensity of a lot of situations, allowing players to gather more information on themselves/teammates/enemies in order to go into battles with better strategy. While some of these are just fluff, many are major QoL that will assist against the shufflemania that is CM by telling players 'hey this person isn't just doing a random do_after, this one is important'.

I am excited for you all to see how brutal new neuro gas is with 0 balance changes.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Blindness:
image

One of brain damage stumbling:
image

Some boiler gas effects:
image
image
image

image

</details>

---

Like all the cool stuff I make?
[ko-fi](https://ko-fi.com/J3J6102QI0)

# Changelog

:cl: GrrrKitten
add: Adds a new section of non-player activateable emotes
add: Adds emotes for getting effected by boiler neuro gas
add: Adds emotes for all forms of organ damage
add: Deaf people can now see visual emotes
qol: Adds emotes for misc occurences that should emote
qol: Stripping/placing items on players now emotes
qol: Surgery, Splinting, Defibbing, and CPR'ing now emotes.
fix: Fixes blindness displaying over runechat
refactor: Minor emote code refactor
/:cl:

### CODE REVIEW

File: code/datums/ammo/bullet/bullet.dm [hunk 1]
```diff
@@ -47,6 +47,8 @@
 		SPAN_HIGHDANGER("[user] aims \the [fired_from] directly at your head!"),
 		SPAN_DANGER("[user] aims \the [fired_from] at [execution_target]'s head!"))
 
+	user.emote("me", 1, "aims [fired_from] at [execution_target]'s head!")
```
Comment (harryob) [inline code/datums/ammo/bullet/bullet.dm:50]: i don't think m_type, and thereby /run_emote(... type_override ...) is even used, why is it 1 here?

Comment (GrrrKitten) [reply code/datums/ammo/bullet/bullet.dm:50]: I honestly couldn’t figure it out after a while of trying and forgot about it, the few emote()’s I saw used it so I assumed it was important, I can mass fix it

**Comment by Tyranicranger4** [comment]
While I like the general idea, a lot of this seems both unnecessary and overly spammable. For example, with its current implementation, here's a clip of what you would see if you're doing surgery at a good pace: https://streamable.com/ysmhyl 

There's also the fact that you can only have one runechat message up at a time, so any of these emotes triggering will immediately bury anything you're trying to say; which is especially unfortunate since a lot of people - myself included - like roleplaying during longer medical steps (surgery, splinting, ect).

Lastly, and this might just be me overthinking things, I feel like having such a wide assortment of frequently triggering automated emotes will make emotes in general more ignorable for a lot of people. Which would be quite a shame.

**Comment by GrrrKitten** [comment]
> While I like the general idea, a lot of this seems both unnecessary and overly spammable. For example, with its current implementation, here's a clip of what you would see if you're doing surgery at a good pace: https://streamable.com/ysmhyl
> 
> There's also the fact that you can only have one runechat message up at a time, so any of these emotes triggering will immediately bury anything you're trying to say; which is especially unfortunate since a lot of people - myself included - like roleplaying during longer medical steps (surgery, splinting, ect).
> 
> Lastly, and this might just be me overthinking things, I feel like having such a wide assortment of frequently triggering automated emotes will make emotes in general more ignorable for a lot of people. Which would be quite a shame.

The surgery steps can be a bit rapid fire but I feel as if the visibility provides more than the emptiness. Also the frequency is not that high on anything else except the visibility

re: the ignorability part, I heavily disagree. CM emotes often feel extremely out of place because if youre emoting that means you are doing it by choice and causes people to look at them like :/, versus this where while it doesn't make emoting overly common, it adds it to some niche things and makes emotes a semi-regular part of gameplay

**Comment by Tyranicranger4** [comment]
> The surgery steps can be a bit rapid fire but I feel as if the visibility provides more than the emptiness.

Emphasis on the part where you can only have one runechat message displaying on a mob at a time. This can and will bury actual messages and emotes, and let's be real here, a lot of newer players don't pay much attention to the actual chat. If I'm trying to roleplay with someone or warn them about something - like say, organ damage - then I don't want "makes an incision on the X's chest" or "applies a splint on the X's chest" to pop up and take priority over my *actual* message.

> versus this where while it doesn't make emoting overly common, it adds it to some niche things and makes emotes a semi-regular part of gameplay

Applying splints, defribrillating people, performing surgery, and suffering from organ damage are very common; these are hardly niche mechanics. These will be seen frequently. Most of the medical item usage could probably be handled better by balloon alerts if you really want to go that route.

> re: the ignorability part, I heavily disagree. CM emotes often feel extremely out of place because if youre emoting that means you are doing it by choice and causes people to look at them like :/, 

This is precisely what I don't like about it. Simple emotes for item usage, while somewhat immersive, do not meaningfully contribute to RP, where as actual emotes do. Someone fluoride staring you because they can't comprehend anyone trying to roleplay on their TDM Aliens server does not mean that the solution is to make emotes so common that they're largely ignorable.

**Comment by zzzmike** [comment]
I feel like this will definitely need a testmerge first to see it how it feels on highpop in a busy area (but I'm looking forward to it)

**Comment by zzzmike** [comment]
On 22977 I played as shipside doc, the emotes felt slightly annoying having them go off on every single surgery step, if you are able to scale them back I think it would be good, but if not I get it.

**Comment by Nomoresolvalou** [comment]
as a synthetic main myself, I find a lot of these emotes annoying. I think the splinting and defib emotes are pretty nice, letting others around you know that you're doing something important, please don't push me. The surgery and suture emotes are the ones I find most awful. I like to use the progress circle as a visual aid for when to begin the next step of a surgery and when to begin suturing another part, but the emotes cover the circle every time, meaning the only way I can tell when I am finished is by either looking at the tiny chat line saying it's done, or by looking at the person I am treating to see when the medical effect goes away. 
Additionally, as a synth, I do shit fast, I can easily finish a surgery step or a suture faster than it takes the emote to go away. It's just a constant stream until I'm done.
Finally, as a sort of cherry on top, that's only one person working. In a hectic environment with a couple other medics or doctors it becomes difficult to even see the person I'm working on because they're covered with emotes. I think the surgery and grafting/suturing emotes should remain as chat box only.

**Comment by Git-Nivrak** [comment]
I've noticed some emotes appear in runechat but not in chat logs, specifically the larva burst message for example

**Comment by harryob** [comment]
you mentioned you had a fix that would close #6666 in that pr, doesn't seem to have been committed here yet. re-open for review when it is

**Comment by Superwelder** [comment]
Ok. I have played a fair few rounds as medic with this PR in testing. To avoid the off topic hammer I will keep this post to how I see this PR conflicts with other established mechanics in CM and exacerbates a particularly annoying bug/oversight I reported a couple months ago. I do not participate on the forums bute I can be reached as Mao on the Discord if someone have a question about my experience with the PR.

First, the PR obviously conveys what medical action is taking place with the above head text but that above head spot is PRIME real estate for one very important circular progress bar when taking any action that requires time, such as splinting or grafting a body part. When an action like using the surgical line repeatedly takes up the above head space, it obfuscates the view of that progress bar. I have found this especially annoying when lining/grafting parts with <30 damage as the above head message does not go away by the time you finish and can move on to the next part.

The other major problem this PR has strongly correlates with  #6242. Generating the above head emote text causes any text you type and enter between the time the the emote was generated and when the emote would naturally expire/disappear to also disappear(the typed message) when the generated emote would have otherwise disappeared. This has the effect of rendering your ability to communicate while performing an action that generates an emote text above head much, much more difficult and clunky. As a medic, this is precisely the time you need to communicate the most. For instance, I have now had multiple cases where I had a patient red lining and I needed to graft their burns before I could defib to revive them. With this nasty combo, I am unable to effectively tell people around me to do CPR because the emote will override the 'CPR' message and worse, even if I stop using the graft to then tell someone to do CPR, I have to wait until the time the emote for grafting disappears before I can enter my plea for CPR because if I do not, the 'CPR' above head message disappears near instantly. Same goes for trying to speak to other medics when I am working to coordinate. As I reported it, that bug/oversight predates this PR by some time yet, this PR procs that bug/oversight more than any other mechanic I have seen in CM.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by harryob** [comment]
pr seems to be stale, let us know if you want it re-opening, or just make a fresh pr for it

**Pull request closed without merging**